.github/workflows/ci.yml

```diff
@@ -179,31 +186,159 @@ jobs:
       - name: 'Init the terraform infrastructure for policy'
         run: |-
-          cp -r ${{ env.TF_MODULE_DIR}} ${{ env.POLICY_TF_MODULE_DIR}}
-          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR}} init
+          cp -r ${{ env.TF_MODULE_DIR }}/* ${{ env.POLICY_TF_MODULE_DIR }}
+          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR }} init
       - name: 'Apply the infrastructure updates for policy'
         id: 'tf_apply_policy'
         run: |-
-          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR}} apply -auto-approve -var-file=${{env.POLICY_TF_VAR_FILE}};
+          terraform -chdir=${{ env.POLICY_TF_MODULE_DIR }} apply -auto-approve -var-file=${{ env.POLICY_TF_VAR_FILE }};
+      - name: 'Upload terraform files for mapping'
+        uses: 'actions/upload-artifact@v3'
+        if: '${{ steps.tf_apply_mapping.outcome == ''success'' }}'
+        with:
+          name: 'terraform-mapping-files'
+          path: |
+            ${{ env.MAPPING_TF_VAR_FILE }}
+            ${{ env.MAPPING_TF_MODULE_DIR }}/terraform.tfstate
+            ${{ env.MAPPING_TF_MODULE_DIR }}/terraform.tfstate.backup
+
+          retention-days: 1
+      - name: 'Upload terraform files for policy'
+        uses: 'actions/upload-artifact@v3'
+        if: '${{ steps.tf_apply_policy.outcome == ''success'' }}'
+        with:
+          name: 'terraform-policy-files'
+          path: |
+            ${{ env.POLICY_TF_VAR_FILE }}
+            ${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate
+            ${{ env.POLICY_TF_MODULE_DIR }}/terraform.tfstate.backup
+          retention-days: 1
+
+  # Call reuseble workflow
+  call_mapping_reusble_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/mapping/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/mapping'
+
+  call_policy_reusble_workflow:
+    needs: ['set_up_integration_infra']
+    uses: 'abcxyz/pmap/.github/workflows/snapshot-file-copy.yml@main' # ratchet:exclude
+    with:
+      workload_identity_provider: 'projects/921163060412/locations/global/workloadIdentityPools/github-pool-3b39/providers/github-provider'
+      service_account: 'pmap-3b39-ci-sa@pmap-ci.iam.gserviceaccount.com'
+      destination_prefix: 'pmap-ci/policy/${{ needs.set_up_integration_infra.outputs.INTEG_TEST_OBJECT_PREFIX_OUTPUT }}'
+      path: 'test/e2e/testdata/fakedata/policy'
+
+  run_integ_test:
+    runs-on: 'ubuntu-latest'
+    needs: ['set_up_integration_infra', 'call_mapping_reusble_workflow', 'call_policy_reusble_workflow']
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    env:
```

Comment:
For the commonly used envs, can we set them at the top level?

Reply:
Moved to top.